### DISCUSSION TITLE
Meta-Gcode language?

### DISCUSSION THREAD

**Discussion by Rickthebig** [OrcaSlicer/OrcaSlicer#6506] [General]
I often see expressions like this in start/end-gcode, layer change, etc.:
{ if layer_z < 3.6 }M140 T0 S[bed_temperature_initial_layer_single] { endif }
(this is only an example).

1. Is there a description of such control structures (conditional execution, loops, etc.) anywhere?
2. Is that kind of control language valid independent of the firmware flavor or is it intended to be used with Klipper only?

Thanks

**Comment by foreachthing** [comment]
These are Slicer dependent. See here for more https://help.prusa3d.com/article/macros_1775

Not sure if OrcaSlicer has such a page...

**Reply by Rickthebig** [reply]
> These are Slicer dependent.

Well, RRF (RepRap Firmware) has own meta-gcode commands that are interpreted by itself (independent of the slicer). Merlin has very limited macro capabilities (see https://marlinfw.org/docs/gcode/M810-M819.html), but I suppose that Klipper is better in this respect. I guess that if the ready-to-go gcode file created by the slicer is manipulated to include such commands that the firmware understands, it should work (correct me if I am wrong).

But coming back to my question, how are those macro commands handled in Orca?

Anyone here who knows if Orca understands such macro commands and how they are handled?
